core/Symbols.h

```diff
@@ -29,6 +29,18 @@ class SerializerImpl;
 }
 class IntrinsicMethod {
 public:
+    // A list of method names that this intrinsic dispatches to.
+    // Used to ensure that calls involving intrinsics are correctly typechecked on the fast path.
+    //
+    // If your intrinsic does not dispatch to another method, the default implementation returns an
+    // empty vector, so you can elide an implementation.
+    //
+    // If your intrinsic cannot declare the list of methods it might dispatch to (like
+    // `<call-with-splat>`, etc.), you will have to edit Substitute.cc.
+    virtual std::vector<NameRef> dispatchesTo() const {
+        return {};
+    }
```

Comment:
Can we move this out-of-line?

(The first non-abstract virtual method in a class is special, and triggers the output of all kinds of information in the compilation unit which it appears; if that method is defined in a header...the information is output in every file where that header is included.  https://itanium-cxx-abi.github.io/cxx-abi/abi.html#vague-vtable )

Reply:
I've re-requested review to make sure I correctly did the thing you were asking for here.